### PR TITLE
Some fixes for issues I ran into

### PR DESCRIPTION
**Pull request by mikerogerz**
Fixes:
* Added ajax to allowMethod.
* Added save action to list of unlockedActions if using Security component.
* If using Csrf component, turn off for save action.
* Check for session started before starting a new one. Trying to start start a session when one already existed was causing an error.

I'm aware that the ajax response method might only apply to my project (which includes an AjaxView), so you can probably leave that one out.

### CODE REVIEW

File: src/Controller/FeedbackController.php [hunk 1]
```diff
@@ -41,7 +46,7 @@ public function beforeFilter(Event $event) {
 	 * @return \Cake\Http\Response|null
 	 */
 	public function save() {
-	    $this->request->allowMethod('post');
+	    $this->request->allowMethod(['post', 'ajax']);
```
Comment (dereuromark) [inline src/Controller/FeedbackController.php:49]: Isnt an ajax just a flavor? I mean usually we post+ajax, so this should be covered.
Whereas get+ajax shouldnt, right?
But I am fine with this.

File: src/Controller/FeedbackController.php [hunk 2]
```diff
@@ -59,7 +64,9 @@ public function save() {
 			$data['name'] = __d('feedback', 'Anonymous');
 		}
 
-		$this->request->session()->start();
+		if (!$this->request->session()->started()) {
```
Comment (dereuromark) [inline src/Controller/FeedbackController.php:67]: This is done internally in start(), so no need to do that again outside.

**Comment by mikerogerz** [comment]
It resulted in a fatal error for me.

On Jan 11, 2018 2:55 PM, "Mark Sch." <notifications@github.com> wrote:

*@dereuromark* commented on this pull request.
------------------------------

In src/Controller/FeedbackController.php
<https://github.com/dereuromark/cakephp-feedback/pull/4#discussion_r161059282>
:

> @@ -59,7 +64,9 @@ public function save() {
 			$data['name'] = __d('feedback', 'Anonymous');
 		}

-		$this->request->session()->start();
+		if (!$this->request->session()->started()) {

This is done internally in start(), so no need to do that again outside.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/dereuromark/cakephp-feedback/pull/4#pullrequestreview-88287183>,
or mute the thread

**Comment by dereuromark** [comment]
I see
Looks like the core behavior is inconsistent and buggy here regarding this: https://github.com/cakephp/cakephp/blob/master/src/Network/Session.php#L309

So looks good to me.

**Comment by dereuromark** [comment]
Thanks a lot!

**Comment by mikerogerz** [comment]
To you too, appreciate all the work you've done with CakePHP.

**Pull request merged**